Makefile

```
@@ -75,7 +75,7 @@ $(OUT)/trivialDAO_storage.tz : voting_period = 11n
 $(OUT)/trivialDAO_storage.tz: src/**
 	# ============== Compiling TrivialDAO storage ============== #
 	mkdir -p $(OUT)
-	$(BUILD_STORAGE) --output-file $(OUT)/trivialDAO_storage.tz src/base_DAO.mligo base_DAO_contract 'default_full_storage(("$(admin_address)": address), ("$(token_address)": address), $(now_val), $(metadata_map), $(ledger), 1000n, 1n, $(quorum_threshold), 2592000n, 1n, $(voting_period))'
+	$(BUILD_STORAGE) --output-file $(OUT)/trivialDAO_storage.tz src/base_DAO.mligo base_DAO_contract 'default_full_storage({storage_data = {admin = ("$(admin_address)" : address); token_address = ("$(token_address)" : address); now_val = $(now_val); metadata_map = $(metadata_map); ledger = $(ledger); quorum_threshold = $(quorum_threshold); voting_period = $(voting_period)}; config_data = {max_quorum = 1000n; min_quorum = 1n; max_period = 2592000n; min_period = 1n}})'
```

```suggestion
  $(BUILD_STORAGE) --output-file $(OUT)/trivialDAO_storage.tz \
    src/base_DAO.mligo base_DAO_contract "default_full_storage( \
      { storage_data = \
        { admin = (\"$(admin_address)\" : address) \
        ; token_address = (\"$(token_address)\" : address) \
        ; now_val = $(now_val) \
        ; metadata_map = $(metadata_map) \
        ; ledger = $(ledger) \
        ; quorum_threshold = $(quorum_threshold) \
        ; voting_period = $(voting_period)\
      } \
      ; config_data = \
        { max_quorum = 1000n \
        ; min_quorum = 1n \
        ; max_period = 2592000n \
        ; min_period = 1n \
        } \
      } \
    })"
```
and something similar should be done with the other two below.

---

Also I noticed now that we use tabs and as well as 2 spaces for indentation, can you please add a commit to only use 2 spaces instead?

---

I am not familiar with Makefile indentation rules, but if I change a tab to 2 spaces or even 4 spaces, the markdown fails, and makefile doesn't compile

---

I see you made these changes, but as you can see from CI they don't work.

The reason being that you need to wrap the whole string in `"` and not `'` for `\` to work properly, aka this:
```
      src/base_DAO.mligo base_DAO_contract 'default_full_storage( \
<...>
        })'
```
should be:
```
      src/base_DAO.mligo base_DAO_contract "default_full_storage( \
<...>
        })"
```

Perhaps this is the same reason why tabs/spaces are creating a problem?
AFAIK you should be able to replace each tab with 2 spaces and it should work.

---

> Also I noticed now that we use tabs and as well as 2 spaces for indentation, can you please add a commit to only use 2 spaces instead?

https://stackoverflow.com/questions/2131213/can-you-make-valid-makefiles-without-tab-characters

---

Oh thanks, I didn't know that.

There are however two places where we use four spaces (hence why I assumed it was ok), in the `test` and `typescript` targets, @PruStephan can you change those lines to use two tabs instead?

I suppose that the reason why that works is that they are "continuation" lines, but let's be consistent and use tabs everywhere.

---

A question about quorum_threshold. I have just rebased on latest master, and there is a change of quorum_threshold type. 
```
type quorum_threshold =
  [@layout:comb]
  { numerator : nat
  ; denominator : nat
  }
```

So the bound checks should check each parameter separately or remove the check at all

---

The `quorum_threshold` has become a fraction, but so have the bounds.

They don't have to check each parameter separately, but as the fraction, you should still check that `quorum_threshold` is lower than `max_quorum_threshold` and bigger than `min_quorum_threshold`.

See how `cmp_qt` is used to do this.